core/running/test-runner.ts

```diff
@@ -70,9 +80,25 @@ export class TestRunner {
             try {
                 await testItem.run(testSetRunInfo.timeout);
                 result = currentTestResults.addTestCaseResult(testItem.testCase.arguments);
+                errorOccurredRunningTest = null;
+                currentTestIndex += 1;
             }
             catch (error) {
                 result = currentTestResults.addTestCaseResult(testItem.testCase.arguments, error);
+                errorOccurredRunningTest = error;
+            }
+
+            // emit onComplete event out of Alsatian if call back has been defined
+            if ( this._onTestCompleteCB ) {
+                this._onTestCompleteCB( {
+                    currentTestIndex: currentTestIndex,
```

Comment:
There is a bug with the recording of this index in that it will not be incremented if the test throws an error. May be better for safety just now to go with `testId: testSetRunInfo.testPlan.testItems.indexOf(testItem) + 1,` as this should always work and mean we don't have to manage another value.

Also if we could add some tests for this bug that'd be awesome :)